### PR TITLE
Restore nextstrain.org PRs

### LINKED ISSUE #1564: reenable nextstrain.org PR action
We should re-enable the action disabled due to Heroku's review apps temporarily not working. Disabled via https://github.com/nextstrain/auspice/commit/db858051b69db63ce725a673f8dd8e7edd3b922a.

Previous behavior was to create a nextstrain.org PR for every auspice PR, but this proved to be a bit excessive. I think it'd be better to make this a manual action, as it's only useful for auspice PRs which may affect consumers such as nextstrain.org. 

@victorlin you're our local GitHub actions expert, would you mind taking a look at this?

### PR DESCRIPTION
**Pull request by victorlin**
### Description of proposed changes

Follow-up to db858051b69db63ce725a673f8dd8e7edd3b922a.

Heroku is building review apps for nextstrain.org now.

### Related issue(s)

- Closes #1564 

### Testing

- [x] Manual trigger using `gh workflow run "Make PRs for Nextstrain projects which depend on Auspice" --repo nextstrain/auspice --ref victorlin/restore-nextstrain.org-prs` succeeds ([link](https://github.com/nextstrain/auspice/actions/runs/3199546724/jobs/5225418043))
- [x] The [resulting PR](https://github.com/nextstrain/nextstrain.org/pull/603) looks good, and a new PR doesn't get created upon updates
- [ ] ~Figure out why runs are taking so long (>15m)~ [not worth it](https://github.com/nextstrain/auspice/pull/1565#issuecomment-1270768343)

### CODE REVIEW

**Comment by tsibley** [comment]
> Heroku is still not building review apps

Uh, yes they are? Maybe we forgot to re-enable them for Auspice, but they work for nextstrain.org.

**Review by tsibley — APPROVED** [review]
+1 for the re-enabling here.

Not the fault of this PR, but I'm still not thrilled that we're passing decently powerful credentials to some random third-party code:

https://github.com/nextstrain/auspice/blob/5ba0b26b7575312e0919eca5f38ab3bf3d0660e5/.github/workflows/make_prs_for_other_repos.yaml#L29-L31

**Comment by jameshadfield** [comment]
Good timing! As per https://github.com/nextstrain/auspice/issues/1564 I'd recommend making this a workflow dispatch rather than automatically making a (nextstrain.org) PR for each auspice PR -- in practice this resulted in too many unnecessary PRs which became confusing.

**Comment by victorlin** [comment]
@tsibley oh you're right, they are working on nextstrain.org PRs. I updated the commit/PR descriptions to reflect that.

**Comment by tsibley** [comment]
@victorlin

From the message of af56b03:

```
Two notes:

1. The title and description of the PR have been updated to use the
   source ref (e.g. branch, tag) rather than PR number, since that is
   not guaranteed or retrievable with the manual trigger.
2. The commit to install Auspice from is changing from the
   GitHub-managed PR merge HEAD to the source ref HEAD. This can have
   implications when run on a PR's source branch if there are notable
   changes in the target branch unincorporated in the source branch. To
   mitigate this, make sure to update the source branch (e.g. with a git
   rebase) before triggering this workflow.
```

To avoid both of these issues, can we instead make the workflow accept a PR number? e.g.:

```
gh workflow run "Make PRs for Nextstrain projects which depend on Auspice" --repo nextstrain/auspice -f pr=1565
```

The workflow would then have the PR number to use and be able to turn `refs/pull/${{inputs.pr}}/merge` into a commit id (SHA) and use that as it previously did.

**Comment by victorlin** [comment]
@tsibley I've opted for automatic retrieval of PR number using `gh pr view`: 9c9bfd32da8c3a1e5059a4ca7854f26c6609b2eb Still WIP but the PR number retrieval should work. Good point about using `refs/pull/${{inputs.pr}}/merge`, I'll add that back.

**Comment by jameshadfield** [comment]
@victorlin is your intention to run these on the head commit of the branch (which is what I did when I set up the action) or the merge commit (`GITHUB_SHA`)? I don't know which is the best for this purpose, but I thought it'd be difficult to try to debug a failing nextstrain.org build if the problem was only introduced by the merge and thus didn't exist on the auspice PR branch.

**Comment by victorlin** [comment]
@jameshadfield oops, I got your original intention backwards in my commit message. You're right, `GITHUB_SHA` for `pull_request` events reflect the GitHub-managed PR merge and your implementation.

In the currently pushed implementation, I intend to run these on the head commit of the branch (not merge), which is simply `GITHUB_SHA` for the `workflow_dispatch` event. So this would be the same as before.

However, @tsibley noted from my (incorrect) commit description that it's still possible to get the merge ref from the PR number. This would change behavior as you mention:

> I thought it'd be difficult to try to debug a failing nextstrain.org build if the problem was only introduced by the merge and thus didn't exist on the auspice PR branch.

I'd actually be in favor of surfacing these failures, since they are bound to come up once actually merged, and better to catch them sooner than later.

**Comment by jameshadfield** [comment]
> I'd actually be in favor of surfacing these failures, since they are bound to come up once actually merged, and better to catch them sooner than later.

Fair enough. Perhaps add a message to this effect in the PR body, as this will catch someone out!

> the head commit of the branch (not merge), which is simply GITHUB_SHA for the workflow_dispatch event

I guess this makes some sense, but I find it super confusing to have GITHUB_SHA refer to different things depending on the trigger. I didn't actually know a branch-merge-commit (which isn't found anywhere in your local git tree) even exists until I wrote the original action here.

**Comment by victorlin** [comment]
Updated to only allow branches with PRs, for simplicity and parity with previous behavior.

**Comment by victorlin** [comment]
This is ready for review again. I went down a deep rabbit hole on the last task in the PR tasklist - figuring out why runs are taking so long [[1]] [[2]], and [trying alternatives unsuccessfully](https://github.com/victorlin/github-test/actions/workflows/test-nextstrain-auspice-install.yml). 15 minute `npm install`s aren't ideal, but I think we can live with it.

[1]: https://github.com/npm/cli/issues/4896
[2]: https://github.com/npm/npm/issues/17246

File: .github/workflows/make_prs_for_other_repos.yaml [hunk 1]
```diff
@@ -1,48 +1,74 @@
...
+      # - auspice-sha: The GitHub-managed merge ref. Used for npm install.
+      - name: Detect PR from branch
+        id: detect-pr
+        run: |
+          PR_NUMBER=$(gh pr view $GITHUB_REF_NAME --repo nextstrain/auspice --json 'number' --jq '.number') || true
+          if [[ -z $PR_NUMBER ]]; then
+            echo "ERROR: This branch is not associated with a PR in Auspice." >&2
+            exit 1
+          fi
+          MERGE_SHA=$(gh pr view $GITHUB_REF_NAME --repo nextstrain/auspice --json 'potentialMergeCommit' --jq '.potentialMergeCommit.oid')
```
Comment (jameshadfield) [inline .github/workflows/make_prs_for_other_repos.yaml:27]: Just double checking I understand what's going on here (via https://github.com/nextstrain/auspice/pull/1565#issuecomment-1269040879) -- the salient thing is that it's a `workflow_dispatch` trigger and so `$GITHUB_SHA` here refers to the head of the branch (which in this case is a PR, but that's not relevant for how the action machinery operates). However, if the action _had_ been triggered by a push-to-branch then `$GITHUB_SHA` would refer to the merge commit SHA?

Perhaps add a one-line comment here explaining?

**Review by jameshadfield — APPROVED** [review]
[Looks good](https://github.com/nextstrain/nextstrain.org/pull/603) @victorlin -- thanks for persevering 😄

**Pull request merged**